tests/Config/InjectedConfigTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Illuminate\Tests\Config;
+
+use Illuminate\Config\Attributes\InjectedConfig;
+use Illuminate\Config\Repository;
+use Illuminate\Container\Container;
+use Mockery as m;
+use PHPUnit\Framework\TestCase;
+
+class InjectedConfigTest extends TestCase
+{
+    private function mockConfigInjection(Repository $mock, $key, $expectedValue): m\LegacyMockInterface|m\MockInterface|Repository|null
+    {
+        return $mock->allows('get')
+            ->with($key)
+            ->andReturns($expectedValue)
+            ->getMock();
+    }
+
+    public function testItInjectsValuesForInjectedConfigAttributedProperties(): void
+    {
+        $container = new Container();
+        $mock = m::mock(Repository::class);
+        $mock = $this->mockConfigInjection($mock, 'foo.string', 'Laravel');
+        $mock = $this->mockConfigInjection($mock, 'foo.boolean', true);
+        $mock = $this->mockConfigInjection($mock, 'foo.int', 1);
+        $mock = $this->mockConfigInjection($mock, 'foo.float', 1.1);
+        $mock = $this->mockConfigInjection($mock, 'foo.array', []);
```

Comment:
I don't see the need to mock all this stuff. It makes the test less readable.

You could just do:

```php
use Illuminate\Config\Repository;

$config = new Repository([
     'foo.string' => 'Laravel',
     // all other params
]);
```

and then bind that to the container

Reply:
Thanks for the suggestion!

Reply:
I have reverted the commits for now because I cannot figure out why this is happening